functionary/ui/views/functions.py

```diff
@@ -0,0 +1,76 @@
+from django.http import HttpResponse, HttpResponseForbidden, HttpResponseRedirect
+from django.shortcuts import render
+from django.views.generic.detail import DetailView
+from django.views.generic.list import ListView
+
+from core.auth import Permission
+from core.models import Environment, Function, Task
+
+from ..forms.forms import FunctionForm
+
+
+class FunctionListView(ListView):
+    model = Function
+
+    def get_queryset(self):
+        """Filters functions not in the environment then sorts based
+        on package name, then function name."""
+
+        env_id = self.request.session["environment_id"]
+        return (
+            super()
+            .get_queryset()
+            .filter(package__environment__id=env_id)
+            .order_by("package__name", "name")
+        )
+
+
+class FunctionDetailView(DetailView):
+    model = Function
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        form: FunctionForm = FunctionForm(self.get_object())
+        context["form"] = form.render("forms/function_detail.html")
+        return context
+
+
+def execute(request) -> HttpResponse:
+    func: Function = None
+    form: FunctionForm = None
```

Comment:
Another type declaration comment...

In this case, you're declaring the type, but then immediately setting the value to something other than that type.  If you had some some sort of linter / static checker here it would almost certainly yell at you.

For example:

![image](https://user-images.githubusercontent.com/86254668/189017280-381dc2c1-b610-49dd-80e0-0cbec6820c69.png)

I am a proponent of type hints, most especially in function / method signatures.  For variable declarations it kind of feels like a judgement call in terms of the value added.  In this case, you can either change this to be something like:

```python
form: Optional[FunctionForm] = None
```

Or just opt to drop the type declaration.